woocommerce/woocommerce
# Add product variation visibility toggle

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://github.com/woocommerce/woocommerce/issues/35790






### How to test the changes in this Pull Request:



1. Find a variable product and grab its id
2. Visit `/wp-admin/admin.php?page=wc-admin&path=/product/{productId}` replace `{productId}` for your variable product's id
3. Under `Options` tab go to `Variations` section
4. In the right most column of each row an eye icon should be shown
5. If the variation status is `publish` then the icon should be a normal eye with color gray.
5.1. If hover it a tooltip with the text `Visible to customers` should be shown.
5.2. When clicking the icon a spiner icon should be shown and when the update is complete the eye icon becomes red and with a cancel line.
5.3. Refreshing the page should show the same state as prev.
6. Same as point 5 in reverse way but with status `private`.

<video src="https://user-images.githubusercontent.com/13334210/207924847-c6ea94d4-098b-43ba-a7f5-5bc16fa97ec0.mov"></video>



### Other information:




### FOR PR REVIEWER ONLY:
